numba/cuda/compiler.py

```diff
@@ -74,6 +71,56 @@ def compile_kernel(pyfunc, args, link, debug=False, inline=False,
     return cukern
 
 
+@global_compiler_lock
+def compile_ptx(pyfunc, args, debug=False, device=False, fastmath=False,
+                cc=None, opt=True):
+    """Compile a Python function to PTX for the given arguments.
+
+    :param pyfunc: The Python function to compile.
+    :param args: A tuple of argument types to compile for.
+    :param debug: Whether to include debug info in the generated PTX.
+    :type debug: bool
+    :param fastmath: Whether to enable fast math flags (ftz=1, prec_sqrt=0,
+                     prec_div=, and fma=1)
+    :type fastmath: bool
+    :param cc: Compute capability to compile for, as a tuple ``(MAJOR, MINOR)``.
+               Defaults to ``(5, 2)``.
+    :type cc: tuple
+    :param opt: Enable optimizations. Defaults to ``True``.
+    :type opt: bool
+    :return: PTX code generated by NVVM.
+    :rtype: bytes
+    """
+    cres = compile_cuda(pyfunc, None, args, debug=debug)
+    resty = cres.signature.return_type
+    if device:
+        llvm_module = cres.library._final_module
+        nvvm.fix_data_layout(llvm_module)
```

Comment:
1. The `device` argument is not documented in the docstring
2. Is this the option for generating relocatable device code (like `-rdc` for nvcc and nvrtc)?

Reply:
Sorry, misread the discussions above 😅 So `device` is for generating device functions (`__device__`). My questions above still stand, though.

Reply:
Thanks - I've added some documentation, and it is indeed for compiling a `device` function. As a result of your question, I did do some experimentation with the `-rdc` flag in https://github.com/gmarkall/ptx-rdc-comparison, which also relates to https://github.com/numba/numba/issues/4574.

Reply:
Thanks, Graham! 🙏 Will take a look later and move the discussion to #4574.